src/control/__tests__/describeIndex.test.ts

```diff
@@ -6,28 +6,47 @@ import {
 } from '../../errors';
 
 describe('describeIndex', () => {
-  const responseData = Object.freeze({
-    database: {
-      name: 'test-index',
-      dimensions: undefined,
-      indexType: undefined,
-      metric: 'cosine',
-      pods: 1,
-      replicas: 1,
-      shards: 1,
-      podType: 'p1.x1',
-      indexConfig: undefined,
-      metadataConfig: undefined,
-    },
-    status: { ready: true, state: 'Ready' },
-  });
+  let responseData;
 
-  test('should remove undefined fields from response', async () => {
-    const IOA = {
+  const setupSuccessResponse = () => {
+    return {
       describeIndex: jest
         .fn()
         .mockImplementation(() => Promise.resolve(responseData)),
     };
+  };
+
+  const setupErrorResponse = (response) => {
+    return {
+      describeIndex: jest
+        .fn()
+        .mockImplementation(() => Promise.reject({ response })),
+      listIndexes: jest
+        .fn()
+        .mockImplementation(() => Promise.resolve(['foo', 'bar'])),
+    };
+  };
+
+  beforeEach(() => {
+    responseData = Object.freeze({
```

Comment:
Yeah, `Object.freeze` doesn't really freeze everything. If there are references to mutable things inside of a frozen object, they can still be mutated. So it's giving a false sense of security and could result in test pollution if anything is mutating these sub-properties.

Contrived example

```js
const x = { foo: 'bar' }
const y = Object.freeze({ x })

// trying to reassign the x property does not work
// and y.x will still have value { 'foo': 'bar' }
y.x = { foo: 'baz' }

// However, mutating the x reference DOES work
// even when y is frozen.
x.foo = 'quux'
// Now y's value is { x: { foo: 'quux' }}
// This is not a violation of the freeze because the object
// reference behind the x property did not change.
```